pkg/cmd/describe/describe_rollout.go

```diff
@@ -0,0 +1,600 @@
+	"github.com/spf13/cobra"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/duration"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/client-go/kubernetes/scheme"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+const (
+	tableFormat = "%-19s%v\n"
+)
+
+var (
+	rolloutLong = templates.LongDesc(i18n.T(`
+		Get details about and visual representation of a rollout.`))
+
+	rolloutExample = templates.Examples(`
+		# Describe the rollout named rollout-demo within namespace default
+		kubectl-kruise describe rollout rollout-demo/default
+
+		# Watch for changes to the rollout named rollout-demo
+		kubectl-kruise describe rollout rollout-demo/default -w`)
+)
+
+type DescribeRolloutOptions struct {
+	genericclioptions.IOStreams
+	Builder          func() *resource.Builder
+	Namespace        string
+	EnforceNamespace bool
+	Resources        []string
+	RolloutViewerFn  func(runtime.Object) (*rolloutsapi.Rollout, error)
+	Watch            bool
+	NoColor          bool
+	TimeoutSeconds   int
+	RolloutsClient   rolloutsv1beta1types.RolloutInterface
+}
+
+type WorkloadInfo struct {
+	Name     string
+	Kind     string
+	Images   []string
+	Replicas struct {
+		Desired   int32
+		Current   int32
+		Updated   int32
+		Ready     int32
+		Available int32
+	}
+	Pod []struct {
+		Name     string
+		BatchID  string
+		Status   string
+		Ready    string
+		Age      string
+		Restarts string
+		Revision string
+	}
+	CurrentRevision string
+	UpdateRevision  string
+}
+
+func NewCmdDescribeRollout(f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := &DescribeRolloutOptions{IOStreams: streams}
+	cmd := &cobra.Command{
+		Use:                   "rollout SUBCOMMAND",
+		DisableFlagsInUseLine: true,
+		Short:                 i18n.T("Get details about a rollout"),
+		Long:                  rolloutLong,
+		Example:               rolloutExample,
+		Aliases:               []string{"rollouts", "ro"},
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(f, args))
+			cmdutil.CheckErr(o.Validate())
+			cmdutil.CheckErr(o.Run())
+		},
+	}
+
+	cmd.Flags().BoolVarP(&o.Watch, "watch", "w", false, "Watch for changes to the rollout")
+	cmd.Flags().BoolVar(&o.NoColor, "no-color", false, "If true, print output without color")
+	cmd.Flags().IntVar(&o.TimeoutSeconds, "timeout", 0, "Timeout after specified seconds")
+
+	return cmd
+}
+
+func (o *DescribeRolloutOptions) Complete(f cmdutil.Factory, args []string) error {
+	var err error
+
+	o.Resources = args
+	o.Namespace, o.EnforceNamespace, err = f.ToRawKubeConfigLoader().Namespace()
+
+	if err != nil {
+		return err
+	}
+
+	parts := strings.Split(args[0], "/")
```

Comment:
the following command will panic due 
kubectl kruise  describe 

panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/openkruise/kruise-tools/pkg/cmd/describe.(*DescribeRolloutOptions).Complete(0x1400023c360, {0x101f22bc0, 0x1400028e280}, {0x102d3d720, 0x0, 0x0?})
	/Users/shouchen/git/kruise-tools/src/github.com/openkruise/kruise-tools/pkg/cmd/describe/describe_rollout.go:137 +0x324